gallery_dl/extractor/generic.py

```diff
@@ -0,0 +1,207 @@
+# -*- coding: utf-8 -*-
+
+"""Extractor for images in a generic web page."""
+
+from .common import Extractor, Message
+from .. import text
+import re
+import os.path
+
+
+class GenericExtractor(Extractor):
+    """Extractor for images in a generic web page."""
+
+    category = "generic"
+    directory_fmt = ("generic_{pageurl}",)
+    filename_fmt = "{filename}.{extension}"
+    archive_fmt = filename_fmt
+
+    # Allow any url, optionally prefixed by "g:" or "generic:"
+    # in case we want to override other extractors.
+    # Based on: https://tools.ietf.org/html/rfc3986#appendix-B
+    pattern = r"""(?ix)
+            (?P<generic>g(?:eneric)?:)?     # optional "g(eneric):" prefix
+            (?P<scheme>https?://)?          # optional http or https scheme
+            (?P<domain>[^/?&#]+)            # required domain
+            (?P<path>/[^?&#]*)?             # optional path
+            (?:\?(?P<query>[^/?#]*))?       # optional query
+            (?:\#(?P<fragment>.*))?$        # optional fragment
+            """
+
+    def __init__(self, match):
+        """Init."""
+        Extractor.__init__(self, match)
+
+        # Allow (and strip) optional "g(eneric):" prefix;
+        # warn about "forced" or "fall-back" mode
+        if match.group('generic'):
+            self.log.warning("Forcing use of generic information extractor.")
+            self.url = match.group(0).partition(":")[2]
+        else:
+            self.log.warning("Falling back on generic information extractor.")
```

Comment:
The idea comes from `youtube-dl`, which always notifies the user when the generic extractor is used automatically as a fallback, with possible sub-optimal results (BTW it also prefixes all log messages with the name of the extractor currently used, which I find another good idea).

I had to use `warning` because using `info` with the default logging level, nothing would be displayed, or am I wrong?